### PR TITLE
Refactor: Use cookiecutter for template generation 

### PR DESCRIPTION
**Pull request by benmezger**
This uses cookiecutter for handling dynamic project choices. Currently it handles the following variables:

- Enable health check - defaults to no
- Enable sentry - defaults to yes
- Use CircleCI - defaults to yes
- Use git - defaults to yes

For running this:

`cookiecutter https://github.com/CheesecakeLabs/django-drf-boilerplate --checkout refactor/cookiecutter`

The state of the CI:

1. Checkout the repository
2. Generate the boilerplate with the default config 
3. Pass the generated boilerplate along the jobs

The `post_gen_project` hook handles any related tasks after generating the template, currently it does the following:

- If `git` is enabled, initialize a new git repository
- If `health_check` is not enabled, remove it from the project
- if `circleci` is not enabled, remove it from the project

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "project_slug": "django_drf_boilerplate",
+  "project_title": "Django DRF Boilerplate",
+  "enable_sentry": "y",
+  "enable_health_check": "n",
+  "use_git": "y",
+  "use_circleci": "y"
```
Comment (fredericojordan) [inline cookiecutter.json]: Won't booleans make more sense?

```suggestion
  "enable_sentry": true,
  "enable_health_check": false,
  "use_git": true,
  "use_circleci": true
```

File: hooks/post_gen_project.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+if '{{ cookiecutter.enable_health_check }}'.lower() == 'n':
+    remove_health_check()
+
+if '{{ cookiecutter.use_circleci }}'.lower() == 'n':
+    remove_circleci()
+
+# Run git as last
+if '{{ cookiecutter.use_git }}'.lower() == 'y':
+    init_git()
```
Comment (fredericojordan) [inline hooks/post_gen_project.py:61]: Updating these accordingly

File: cookiecutter.json (hunk 1 above)
Comment (benmezger) [reply cookiecutter.json]: Cookiecutter doesn't work well if bools :( https://github.com/cookiecutter/cookiecutter/pull/548

Comment (fredericojordan) [reply cookiecutter.json]: Unless you're planning for more than 2 options on each of those... 🤔

**Review by fredericojordan — APPROVED** [review]
Pretty cool! 👍 
I'd just look into the boolean values...

File: cookiecutter.json (hunk 1 above)
Comment (fredericojordan) [reply cookiecutter.json]: > Cookiecutter doesn't work well if bools :( cookiecutter/cookiecutter#548

Oh 😢

Comment (benmezger) [reply cookiecutter.json]: Unreliable at the moment, see: https://github.com/cookiecutter/cookiecutter/pull/848

File: {{ cookiecutter.project_slug }}/README.md [hunk 3]
```diff
@@ -11,6 +11,13 @@ Currently supporting only Python 3.x.
 **IMPORTANT**:
 Docker Compose is used _just_ for development environment. The Dockerfile works without it.
```
Comment (natamvo) [inline {{ cookiecutter.project_slug }}/README.md:13]: maybe we can have a README outside the folder `{{cookiecutter.project_slug}}` with instructions to clone and use the boilerplate and in this file all about the future project (like a model), is it possible?

**Review by natamvo — COMMENTED** [review]
I think there are more places to check, like terraform variables `project_name`

**Comment by benmezger** [comment]
Hey @CheesecakeLabs/backend, CI is now working :)

**Review by phsetti — APPROVED** [review]

**Review by natamvo — APPROVED** [review]

**Comment by natamvo** [comment]
@benmezger can you fix the conflict files?

**Comment by benmezger** [comment]
These files will still be in conflict, although they are synced. AFAIK, this branch will need to become the new main branch.

**Review by rkpinheiro — APPROVED** [review]

**Review by rodolfolottin — APPROVED** [review]

**Review by karranb — APPROVED** [review]
Awesome ! ! !

**Review by matpfernandes — APPROVED** [review]

**Pull request merged**